torch/testing/_internal/common_utils.py

```diff
@@ -1736,12 +1740,40 @@ def random_sparse_matrix(rows, columns, density=0.01, **kwargs):
     column_indices = [i % columns for i in range(nonzero_elements)]
     random.shuffle(column_indices)
     indices = [row_indices, column_indices]
-    values = torch.randn(nonzero_elements, dtype=dtype, device=device)
-    # ensure that the diagonal dominates
-    values *= torch.tensor([-float(i - j)**2 for i, j in zip(*indices)], dtype=dtype, device=device).exp()
+    if dtype in integral_types():
+        requires_grad = False
+        if values_domain is None:
+            low, high = (-100, 100)
+        else:
+            low, high = values_domain
+
+        low = round(max(low, torch.iinfo(dtype).min))
+        high = round(min(high, min(torch.iinfo(dtype).max, 100)))  # clamp large values to 100
+
+        values = torch.randint(low, high, size=(nonzero_elements, ), dtype=dtype, device=device)
+    elif dtype == torch.bool:
+        requires_grad = False
+        values = torch.randint(0, 2, size=(nonzero_elements, ), dtype=dtype, device=device)
+    else:
+        if values_domain is None:
+            values = torch.randn(nonzero_elements, dtype=dtype, device=device)
+            # ensure that the diagonal dominates
+            if diag_dominates:
+                # this wont work for all dtypes
+                # e.g. "exp" "_vml_cpu" not implemented for 'Half'
+                values *= torch.tensor([-float(i - j)**2 for i, j in zip(*indices)], dtype=dtype, device=device).exp()
+        else:
+            low, high = values_domain
+            high = min(high, 100)  # if high=inf, clamp to 100
+            values = (high - low) * torch.rand(nonzero_elements, dtype=dtype, device=device) + low
+
     indices_tensor = torch.tensor(indices)
-    A = torch.sparse_coo_tensor(indices_tensor, values, (rows, columns), device=device)
-    return A.coalesce()
+    A = torch.sparse_coo_tensor(
+        indices_tensor, values, (rows, columns), device=device, requires_grad=requires_grad
+    )
+    if not uncoalesced:
```

Comment:
In the future I'd like to better consider how to generate sparse tensors, especially how to handle coalescing (or not) and the density target